modules/stochastic_tools/src/transfers/SamplerTransfer.C

```diff
@@ -94,23 +94,27 @@ SamplerTransfer::execute()
 void
 SamplerTransfer::initializeToMultiapp()
 {
-  _samples = _sampler_ptr->getSamples();
+  _samples = _sampler_ptr->getLocalSamples();
   _global_index = _sampler_ptr->getLocalRowBegin();
 }
 
 void
 SamplerTransfer::executeToMultiapp()
 {
-
   SamplerReceiver * ptr = getReceiver(processor_id());
-  std::vector<Real> row = getRow(_global_index);
+
+  std::vector<Real> row;
+  getLocalRow(_global_index - _sampler_ptr->getLocalRowBegin(), row);
+
   ptr->transfer(_parameter_names, row);
-  _global_index += 1;
+
+  _global_index++;
 }
 
 void
 SamplerTransfer::finalizeToMultiapp()
 {
+  _samples.resize(0, 0);
```

Comment:
why to resize?

Reply:
frees up the memory, I don't want to keep copies of the sampled data around. This is only in batch mode, where the sample data is generated in initialize, utilized in execute, and then cleaned up in finalize